### ISSUE TITLE
AuthentificationManager try to authenticate several times through the same provider

### ISSUE THREAD

**Issue by bdecarne** [silexphp/Silex#655]
Hi there,

I am faced with an unexpected behaviour of the SecurityProvider.

Take this service registring, wich use a custom authentification provider for a secured area : 

``` php
$this->register(
    new SecurityServiceProvider(),
    array(
        'security.role_hierarchy' => array(
            'ROLE_SUPER_ADMIN' => array('ROLE_ADMIN'),
            'ROLE_ADMIN' => array('ROLE_USER')
        ),
        'security.firewalls' => array(
            'secured' => array(
                'form' => array(
                    'login_path' => "/user/login",
                    'check_path' => "/user/dologin",
                    "default_target_path" => "/",
                    "always_use_default_target_path" => true,
                    'username_parameter' => 'login[username]',
                    'password_parameter' => 'login[password]',
                    "csrf_parameter" => "login[_token]",
                    "failure_path" => "/user/login",
                ),
                'pattern' => '^/',
                'logout' => array(
                    'logout_path' => "/user/logout",
                    "target" => '/',
                    "invalidate_session" => true,
                    "delete_cookies" => array()
                ),
                'users' => $app['user.repository']
            )
        ),
        'security.authentication_provider.secured.dao' => $app->share(function () use ($app) {
            // return custom authentification provider (webservice call)
        }),
        'security.access_rules' => array(
            array('^/user/login', 'IS_AUTHENTICATED_ANONYMOUSLY'),
            array('^/user/password', 'IS_AUTHENTICATED_ANONYMOUSLY'),
            array('^.*$', 'IS_AUTHENTICATED_FULLY'),
        ),
    )
);
```

In the SecurityServiceProvider, during the _security.firewall_map_ constitution, there is two 'security.authentication_listener.factory.*' called : _security.authentication_listener.factory.form_ AND _security.authentication_listener.factory.logout_. For both of them, the same providerId is provided as the first element of the returned array  : _security.authentication_provider.secured.dao_.

So the same provider is added twice to the provider list _security.authentication_providers_ . This list is used by the manager to try authentification. 

In conclusion, when the authentification failed one time, the anthentification manager try a second time to use the same provider. It's a problem in my situation.

I fix it in the provider to prevent multiple call to the webservice, but it seems to be a unexpected behaviour.
